src/docfx/build/toc/parser/MarkdownTocMarkup.cs

```diff
@@ -111,7 +116,7 @@ TableOfContentsItem GetItem(HeadingBlock block)
                 {
                     var invalidTocSyntaxContent = tocContent.Substring(block.Span.Start, block.Span.Length);
                     errors.Add(Errors.InvalidTocSyntax(new SourceInfo<string>(invalidTocSyntaxContent, block.ToSourceInfo(file: filePath)), "multiple inlines in one heading block is not allowed"));
-                    return currentItem;
+                    return null;
```

Comment:
what could go wrong if add an empty instance of `TableOfContentsItem` into `items`?

Reply:
It adds an empty entry to the TOC node tree, and triggers [name check](https://github.com/dotnet/docfx/pull/4735/files#diff-d483b592e55ca934337e0e273921ee1cR153), results in syntax error and name missing error on the same node.

It does not have this problem before because there was a `tocModelItem.Name != null` check that happened to by-pass this case

cc @OsmondJiang

Reply:
I see